lib/src/configuration.rs

```diff
@@ -0,0 +1,53 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
+
+pub type BasicAuth = (String, Option<String>);
```

Comment:
For consistency with `ApiKey`, we could be more explicit and use a `struct` here with members `username` and `password` (or `user` and `pass`).